compare_output.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# Check Configuration is supplied
+if [[ -z "$SFTP_USER" ]]; then
+    echo "Please provide the SFTP_USER."
+    exit 1
+fi
+
+if [[ -z "$SFTP_SERVER" ]]; then
+    echo "Please provide the SFTP_SERVER"
+    exit 1
+fi
+
+if [[ -z "$REMOTE_PATH" ]]; then
+    echo "Please provide the REMOTE_PATH"
+    exit 1
+fi
+
+LOCAL_DIR="./downloaded_files"
+LOG_FILE="./snos_run.log"
+
+# Create local directory if it does not exist
+mkdir -p "$LOCAL_DIR"
+
+# Start the SFTP session to list files
+# echo "Starting SFTP session to list files..." | tee -a "$LOG_FILE"
+# sftp $SFTP_USER@$SFTP_SERVER << EOF > temp_file_list.txt
+# cd $REMOTE_PATH
+# ls
+# EOF
```

Comment:
Eww, it looks like sftp doesn't make it easy to list directory contents. Maybe an alternative is available (`scp` or `ssh`) where you could do this as part of the script itself?